Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.Dynamics.Nav.CodeAnalysis;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Syntax;
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace BusinessCentral.LinterCop.Design
+{
+    [DiagnosticAnalyzer]
+    public class Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create<DiagnosticDescriptor>(DiagnosticDescriptors.Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable);
+
+        public override void Initialize(AnalysisContext context)
+            => context.RegisterSymbolAction(new Action<SymbolAnalysisContext>(this.CheckTablePrimaryKeyIsNotAutoIncrement), SymbolKind.Table);
+
+        private void CheckTablePrimaryKeyIsNotAutoIncrement(SymbolAnalysisContext context)
+        {
+            ITableTypeSymbol tableTypeSymbol = (ITableTypeSymbol)context.Symbol;
+            if (!IsSymbolAccessible(tableTypeSymbol))
+                return;
+
+            CheckTable(tableTypeSymbol, ref context);
+        }
+
+        private void CheckTable(ITableTypeSymbol table, ref SymbolAnalysisContext context) {
+            if (table.TableType != TableTypeKind.Temporary)
+                return;
+
+            foreach (var field in table.PrimaryKey.Fields) {
+                if (field.GetProperty(PropertyKind.AutoIncrement) != null) {
+                    context.ReportDiagnostic(
+                        Diagnostic.Create(
+                            DiagnosticDescriptors.Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable,
+                            field.GetLocation(),
+                            new object[] { GetDeclaration(field), field.Name }));
```

Comment:
the GetDeclaration is not needed. I also would not give any detailed information here. The location of the diagnostig should be enough. Then you can remove the 3rd parameter of Diagnostig.Create() completely and remove the {0} and {1} from the format string of the diagniostic

Reply:
Ok! :)

Reply:
@StefanMaron 
Should now be done in 
403c90336208a9c788f662101ee9dcfd73b68ed1